components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1661,6 +1741,31 @@ void RewardsServiceImpl::ResetState(
                      AsWeakPtr(), std::move(callback)));
 }
 
+void RewardsServiceImpl::ResetTheWholeState(
+    const base::Callback<void(bool)>& callback) {
+  reset_states_ = true;
+  notification_service_->DeleteAllNotifications();
+  std::vector<base::FilePath> paths;
+  paths.push_back(ledger_state_path_);
```

Comment:
@NejcZdovc Not for this PR but ideally we should be storing all ledger state in `Default/rewards_service` so that when clearing state we can just delete one folder, so if future files we do not need to add to the list of paths